libvast/src/format/zeek.cpp

```diff
@@ -276,8 +277,18 @@ caf::error reader::read_impl(size_t max_events, size_t max_slice_size,
   size_t produced = 0;
   // Loop until reaching EOF or the configured limit of records.
   while (produced < max_events) {
-    // Advance line range and check for EOF.
-    lines_->next();
+    // Advance line range and check for EOF. (In the first iteration, the
+    // current line is the last line of the header)
+    // If we already have some events, set a read timeout to ensure downstream
```

Comment:
Is this an optimization only or do we get wrong behavior if we call finish without adding any data beforehand?

Reply:
Yes, if the reader returns `0` the source will get stuck (see the TODO in source.hpp) I stopped my investigation into *why* exactly this happens, since I think @dominiklohmann already understands the bug.

But I think even without that bug it makes sense to keep this logic, for consistency with how we also block indefinitely when awaiting header lines.

Reply:
It's a upstream bug in CAF that's causing this failure. Working around it is no small refactoring on our end. Continuous stream sources cannot be stopped when they've produced 0 results, because the done handler is not called when calling advance on the stream source. We're unable to yield control back to the parent actor, which is why the delayed_send approach mentioned in the comment won't work.

It's a good thing you've stopped investigating this. 😅

Reply:
Since the source is unusable after it returns 0 once, how about just terminating the source so that we at least get an error?!

Reply:
It isn't strictly unusable, it simply locks until new input is available. We can't know for sure at that point if we're stalling forever, or if new input will soon come.

Let's discuss this on the whiteboard at some point in the future. I've debugged this thoroughly just a few weeks back.